File: C++/src/Solution201_300/Solution217.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution217.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include <unordered_set>

bool Solution217::containsDuplicate(vector<int> &nums) {
    unordered_set<int> hashset;
    for (int i = 0; i < nums.size(); i++){
        if (hashset.count(nums[i])){
            return true;
        }
        hashset.insert(nums[i]);
    }
    return false;
}

File: Java/src/main/java/QuestionBank/Solution201_300/Solution217.java
package QuestionBank.Solution201_300;

import java.util.HashSet;

/**
 * @author windows
 */
public class Solution217 {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for (int i = 0; i < nums.length; i++) {
            if (set.contains(nums[i])) {
                return true;
            }else{
                set.add(nums[i]);
            }
        }
        return false;
    }


    public static void main(String[] args) {
        Solution217 solution217 = new Solution217();
    }
}